mlir/utils/performance/analysis/compileAndCollectTuningData.py

```
@@ -0,0 +1,604 @@
+                        'sgpr_spills': result_dict.get('sgpr_spills', ''),
+                        'LDS_allocated': result_dict.get('LDS_allocated', ''),
+                        'occupancy': result_dict.get('occupancy', ''),
+                        'WFsPerWG': result_dict.get('WFsPerWG', ''),
+                        'mfma_wmma_instruction': result_dict.get('mfma_wmma_instruction', '')
+                    }
+                
+                writer.writerow(row)
+        
+        print(f"\nResults written to {output_file}")
+        
+    except Exception as e:
+        print(f"\nError writing results to tsv: {e}")
+
+def print_progress(current, total):
+    """Print a progress bar to stdout."""
+    prefix = "Processing Configs"
+    percent = (current / total) * 100
+    bar_length = 40
+    filled_length = int(bar_length * current // total)
+    bar = '█' * filled_length + '-' * (bar_length - filled_length)
+    print(f'\r{prefix}: |{bar}| {current}/{total} ({percent:.1f}%)', end='',
+          flush=True)
+    if current == total:
+        print()  # New line when complete
+
+def main():
+    """Main function to process configurations and collect tuning data."""
+    parser = argparse.ArgumentParser(
+        description="Compile configurations and collect tuning data",
+        formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    parser.add_argument('--op', required=True,
+                        help='Operation to perform (e.g., "compile")',
+                        choices=['conv', 'gemm', 'attention'])
+    parser.add_argument('config_tsv', help='Path to the tuning database file')
+    
+    args = parser.parse_args()
+
+    # Get the paths to the rocmlir binaries
+    build_bin_dir = os.path.dirname(os.path.abspath(__file__))
+    rocmlir_root = os.path.dirname(build_bin_dir)
+    paths = perfRunner.create_paths(None, rocmlir_root)
+
+    # Parse the configuration file
+    configs = perfRunner.read_tuning_db(args.config_tsv, True)
```

(if we intend to run this in CI (to avoid silent errors in ci)) 
check if config_tsv file exists and return non zero if it doesn't